pandas/indexes/multi.py

```diff
@@ -2064,20 +2066,69 @@ def difference(self, other):
             return self
 
         if self.equals(other):
-            return MultiIndex(levels=[[]] * self.nlevels,
-                              labels=[[]] * self.nlevels,
-                              names=result_names, verify_integrity=False)
+            return self._create_as_empty(names=result_names)
 
         difference = sorted(set(self._values) - set(other._values))
 
         if len(difference) == 0:
-            return MultiIndex(levels=[[]] * self.nlevels,
-                              labels=[[]] * self.nlevels,
-                              names=result_names, verify_integrity=False)
+            return self._create_as_empty(names=result_names)
         else:
             return MultiIndex.from_tuples(difference, sortorder=0,
                                           names=result_names)
 
+    def _create_as_empty(self, nlevels=None, names=None,
+                         verify_integrity=False):
+        """
+        Creates an empty MultiIndex
+
+        Parameters
+        -------
+        nlevels : optional int, default None
+            The number of levels in the empty MultiIndex. If None defaults
+            to the current number of levels
+        names : optional sequence of objects, default None
+            Names for each of the index levels. If None defaults to the
+            current names.
+        verify_integrity : boolean, default False
+            Check that the levels/labels are consistent and valid
+
+        Returns
+        -------
+        empty : MultiIndex
+        """
+
+        if nlevels is None:
+            nlevels = len(self.levels)
+        if names is None:
+            names = self.names
+
+        return MultiIndex(levels=[[]] * nlevels,
+                          labels=[[]] * nlevels,
+                          names=names, verify_integrity=verify_integrity)
+
+    def symmetric_difference(self, other, result_name=None):
+        """
+        Compute sorted set symmetric difference of two MultiIndex objects
+
+        Returns
+        -------
+        diff : MultiIndex
+        """
+        self._assert_can_do_setop(other)
+        other, result_name_update = self._convert_can_do_setop(other)
+
+        if result_name is None:
+            result_name = result_name_update
+
+        if self.equals(other):
+            return self._create_as_empty(names=result_name)
+
+        difference = sorted(set((self.difference(other)).
+                                union(other.difference(self))))
```

Comment:
Just a comment:

This sorting breaks with mixed-int values (same issue as in #13432). I suppose this line may be safely replaced with the code from PR #13514, slightly changed, namely:
```python
        this = self._get_unique_index()
        other = other._get_unique_index()
        indexer = this.get_indexer(other)

        # {this} minus {other}
        common_indexer = indexer.take((indexer != -1).nonzero()[0])
        left_indexer = np.setdiff1d(np.arange(this.size), common_indexer,
                                    assume_unique=True)
        left_diff = this.values.take(left_indexer)
 
        # {other} minus {this}
        right_indexer = (indexer == -1).nonzero()[0]
        right_diff = other.values.take(right_indexer)

        the_diff = _concat._concat_compat([left_diff, right_diff])
```
Then if `the_diff` is empty, an empty MI should be returned, otherwise:
```python
        return self._shallow_copy(the_diff, names=result_name).sortlevel()
```
(In comparison to `Index.symmetric_difference` sorting is postponed until the end.)

Similar change may be applied to `MultiIndex.difference`, I guess.